libcaf_core/src/actor_system_config.cpp

```diff
@@ -367,22 +377,23 @@ std::string actor_system_config::render_pec(uint8_t x, atom_value,
                         meta::omittable_if_empty(), xs);
 }
 
-void actor_system_config::extract_config_file_path(string_list& args) {
+error actor_system_config::extract_config_file_path(string_list& args) {
   auto ptr = custom_options_.qualified_name_lookup("global.config-file");
   CAF_ASSERT(ptr != nullptr);
   string_list::iterator i;
   string_view path;
   std::tie(i, path) = find_by_long_name(*ptr, args.begin(), args.end());
   if (i == args.end())
-    return;
+    return {};
```

Comment:
It's more idiomatic (and easier to comprehend IMO) to write `return none` instead.